tasks/main.yml

```diff
@@ -36,29 +36,14 @@
 - name: install docker-rotate cron script
   template: src=docker-rotate.j2 dest=/etc/cron.daily/docker-rotate mode=755
 
-- name: push logrotate config file for docker container json logs
-  template: src=logrotate.conf.j2 dest=/etc/logrotate.d/docker
-  when: installed_docker_version | version_compare('1.8', '<')
-
+# docker log rotation is now managed by Docker (since Docker 1.8). We no longer deal with
+# it in this role. Clean up any old file that might still be around.
 - name: make sure docker logrotate config is removed
   file: state=absent dest=/etc/logrotate.d/docker
-  when: installed_docker_version | version_compare('1.8', '>=')
 
 - name: install docker zombie volumes cleanup cron script
   copy: src=clear-zombie-docker-data-dirs dest=/etc/cron.daily/ mode=755
 
-- name: push docker configuration
-  template: src=docker.j2 dest=/etc/default/docker
-  register: docker_config_result
-  notify: restart docker
-
-# make sure docker restarts to get the newest configuration
-- meta: flush_handlers
```

Comment:
From what I could see, now that we've removed the configuration code, docker-base is no longer doing anything that would require it to restart docker. AFAIK the "flush_handlers" here is just to make sure that the docker restart happens.(Though, to be fair, it could have other side effects.)

The old flow was:
 - "docker" role would install docker and make sure it was running
 - "docker" role would reload ansible facts
 - "docker-base" role would manage docker config and, if it changed, notify "restart docker"
 - "docker-base" role would flush handlers
 - "docker-base" role would reload ansible facts

The new flow is:
 - "docker" role writes docker config
 - "docker" role installs or updates docker if required
 - "docker" role restarts docker if config was updated and docker wasn't
 - "docker" role reloads ansible facts
Note one less restart!